### PR TITLE
feat(catalog.tracking): data tracking

### PR DESCRIPTION
**Pull request by Kranysys**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/EPIC-MANAGER-11852`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-11856
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

## Description

Adding data tracking to the catalog revamp

### CODE REVIEW

File: packages/manager/apps/catalog-revamp/src/components/Filters/FilterItem.tsx [hunk 1]
```diff
@@ -35,6 +35,7 @@ const FilterItem: React.FC<FilterItemProps> = ({
         onOdsCheckedChange={(event: CustomEvent) =>
           onCheckboxChange(type, label, event)
         }
+        data-tracking={`manager_product_catalog::filter::${type}::${label}`}
```
Comment (aboungnaseng-ovhcloud) [inline packages/manager/apps/catalog-revamp/src/components/Filters/FilterItem.tsx:38]: you don't need to add manager_product_catalog

Comment (Kranysys) [reply packages/manager/apps/catalog-revamp/src/components/Filters/FilterItem.tsx:38]: Done, removed.

File: packages/manager/apps/catalog-revamp/src/components/SearchBar/SearchBar.tsx [hunk 2]
```diff
@@ -57,6 +63,7 @@ const SearchBar: React.FC<SearchbarProps> = ({
             variant={ODS_BUTTON_VARIANT.stroked}
             color={ODS_THEME_COLOR_INTENT.primary}
             onClick={() => setShowFilters((filterState) => !filterState)}
+            data-tracking={showFilters ? `filter::show` : `filter::hide`}
```
Comment (chipp972) [inline packages/manager/apps/catalog-revamp/src/components/SearchBar/SearchBar.tsx:66]:
```suggestion
            data-tracking={`filter::${showFilters ? 'show' : 'hide'}`}
```

Comment (Kranysys) [reply packages/manager/apps/catalog-revamp/src/components/SearchBar/SearchBar.tsx]: Done, Thanks

**Review by chipp972 — APPROVED** [review]

**Pull request merged**